server/app/tasks/DatabaseSeedTask.java

```diff
@@ -91,33 +93,45 @@ public DatabaseSeedTask(QuestionService questionService, VersionRepository versi
     this.database = DB.getDefault();
   }
 
-  public void run() {
-    seedCanonicalQuestions();
+  public ImmutableList<QuestionDefinition> run() {
+    return seedCanonicalQuestions();
   }
 
   /**
    * Ensures that questions with names matching those in {@code CANONICAL_QUESTIONS} are present in
    * the database, inserting the definitions in {@code CANONICAL_QUESTIONS} if any aren't found.
    */
-  private void seedCanonicalQuestions() {
-    if (canonicalQuestionsAlreadyPresent()) {
-      return;
+  private ImmutableList<QuestionDefinition> seedCanonicalQuestions() {
+    ImmutableSet<String> canonicalQuestionNames =
+        CANONICAL_QUESTIONS.stream()
+            .map(QuestionDefinition::getName)
+            .collect(ImmutableSet.toImmutableSet());
+    ImmutableMap<String, QuestionDefinition> existingCanonicalQuestions =
+        questionService.getExistingQuestions(canonicalQuestionNames);
+    if (existingCanonicalQuestions.size() < canonicalQuestionNames.size()) {
+      // Ensure a draft version exists to avoid transaction collisions with getDraftVersion.
+      versionRepository.getDraftVersion();
     }
```

Comment:
suggestion (non-blocking): could this be an early return with the boolean inverted instead?